### DISCUSSION TITLE
uml diagram editor

### DISCUSSION THREAD

**Discussion by ms10895** [eclipse-glsp/glsp#1025] [Q&A] [answered]
Hi,
I’m new to glsp and I’m trying to create editors for package diagram and class diagram and I want to use uml2 metamodel as source model.

My question is do i need to implement two separates glsp clients for each diagram editors and maybe a common glsp server.?

which combination would be best for this kind of senario? 

Java-emf-Theia (https://github.com/eclipse-glsp/glsp-examples/tree/master/project-templates/java-emf-theia)
or
Java-emf.cloud-theia(https://github.com/eclipse-emfcloud/modelserver-glsp-integration/tree/main/project-templates/modelserver-glspjava-emf-theia) . can you please give me any ideas or suggestion?

thanks.

**Comment by PatrikN-work** [comment]
Not exactly whay you are asking but have you seen this? https://github.com/borkdominik/bigUML

**Comment by ndoschek** [comment, marked as answer]
Hi @ms10895,

Yes, exactly, as @PatrikN-work already pointed out (thanks!), there are a few projects that use the uml2 metamodel.
- https://github.com/borkdominik/bigUML
- https://github.com/eclipsesource/uml-glsp (this was the base work for the first one)

Both are based on the [modelserver-glspjava-emf-theia template](https://github.com/eclipse-emfcloud/modelserver-glsp-integration/tree/main/project-templates/modelserver-glspjava-emf-theia).
However, they are not up-to-date regarding the dependencies of the used frameworks.

If there are any obstacles, just let us know.
HTH and BR,
Nina
